custom/enikshay/integrations/utils.py

```diff
@@ -21,15 +21,26 @@ def _is_submission_from_test_location(case_id, owner_id):
 
 
 def is_valid_person_submission(person_case):
-    return not _is_submission_from_test_location(person_case.case_id, person_case.owner_id)
+    owner_id = person_case.owner_id
+    if owner_id == ARCHIVED_CASE_OWNER_ID:
+        owner_id = person_case.dynamic_case_properties.get("last_owner")
+    return not _is_submission_from_test_location(person_case.case_id, owner_id)
```

Comment:
Ha, just saw https://github.com/dimagi/commcare-hq/pull/15998/files, so looks like I do need to change this :)